pkg/controller/controller.go

```diff
@@ -111,6 +111,18 @@ func (c *controller) serviceBindingCallback(e k8swatch.Event) error {
 			return err
 		}
 		glog.Infof("Created Service Binding: %s\n%v\n", sb.Name, created)
+	} else if e.Type == k8swatch.Deleted {
+		// if we've already processed the deletion event, ignore
```

Comment:
I don't follow the whole logic, but this smells like a race to me. How are we going to get something that is already deleted? Does this have to do with the whole 'periodic list vs continuous watch' paradigm?